### ISSUE TITLE
Translation fails when C `extern` arrays are promoted to pointers

### ISSUE THREAD

**Issue by brianhuffman** [GaloisInc/crucible#8]
If the following C file
```
extern int arr[2];

int deref(int *x) {
	return x[0];
}

int test() {
	return deref(arr);
}
```
 is compiled to LLVM and run through crucible (using `crucible-integration` branch of saw-script) we get the following error:
```
at internal: unsupported LLVM value: ValNull of type [2 x i32]
```

If the `extern` keyword is removed, then the example works.

**Comment by brianhuffman** [comment]
Looking at the llvm-pretty `Module` values produced in either case, one of the `Global` values inside is affected by the `extern` declaration. Without `extern`, the `gaLinkage` field of the `globalAttrs` field is `Just Common`, but with `extern` it is `Nothing`. The other difference is that without `extern`, the `globalValue` field is `Just ValZeroInit`, but with `extern` it is `Just ValNull`. This seems to be the `ValNull` that causes the error message.

The source of the `ValNull` looks like it is the `finalizeGlobal` function from `Data.LLVM.BitCode.IR.Globals` in the `llvm-pretty-bc-parser` package:
```
finalizeGlobal :: PartialGlobal -> Parse Global
finalizeGlobal pg = case pgValueIx pg of
  Nothing -> mkGlobal ValNull
  Just ix -> do
    tv <- getFnValueById (pgType pg) (fromIntegral ix)
    mkGlobal =<< relabel (const requireBbEntryName) (typedValue tv)
  where
  mkGlobal val =
    do md <- mapM (relabel (const requireBbEntryName)) (pgMd pg)
       return Global { globalSym   = pgSym pg
                     , globalAttrs = pgAttrs pg
                     , globalType  = pgType pg
                     , globalValue = Just val
                     , globalAlign = pgAlign pg
                     , globalMetadata = md
                     }
```
In particular, it looks like the `Nothing -> mkGlobal ValNull` line is probably the source of the `ValNull`. Is this a bug?

**Comment by brianhuffman** [comment]
I talked with @elliottt about it, and we decided that `finalizeGlobal` should be changed. Fixed in GaloisInc/llvm-pretty-bc-parser@90e5b1b6.
